code42_connector.py

```diff
@@ -28,6 +31,19 @@ def __init__(self):
         self._username = None
         self._password = None
         self._client = None
+        self._action_map = {
+            self.TEST_CONNECTIVITY_ACTION_ID: lambda x: self._handle_test_connectivity(x),
+            self.ADD_DEPARTING_EMPLOYEE_ACTION_ID: lambda x: self._handle_add_departing_employee(x),
+            self.REMOVE_DEPARTING_EMPLOYEE_ACTION_ID: lambda x: self._handle_remove_departing_employee(x)
+        }
+
+    @property
+    def client(self):
```

Comment:
I'm pretty sure that every action just calls `main()` and will end up creating a new Connector, so a new client will always end up getting created on every action.

Reply:
Yes, but there are helper methods that may also need to use the client, such as the method for getting the ID.. So rather than assuming the SDK is initialized at the start of every method, this make it so that it will just happen the first time it is used.

Reply:
Oh, and I don't think the logic in __main__ is anything except for debugging actions, from what I can tell.

In a Playbook setting, I am not sure if the client actually gets re-initialized per action.

Reply:
> In a Playbook setting, I am not sure if the client actually gets re-initialized per action.

If it's not through `main()`, then I have no idea how Phantom could possibly discover the `Code42Connector` class and use it

Reply:
The reason I think this is because in this article: https://docs.splunk.com/Documentation/Phantom/4.10.3/DevelopApps/Tutorial

The section titled `Debugging` is the only place from what I can tell where `maint()` is mentioned.. Also, notice how it imports `pudb` at the top of the method. There is no reason a debugger module would be needed in production.

Here is quote from the article: `Running the ipinfoio_connector.py script under a debugger runs the app in standalone mode. In other words, for the most part, it does not talk with the Splunk Phantom platform. Any progress messages that the app sends are dumped on the console and not sent to the platform. This type of execution helps in debugging the app code that interacts with the external device, in this case the ipinfo.io servers.`

That sounds like typically connectors are not run in standalone mode.

I could be wrong, but just sharing why I think this.